src/s1reader/s1_annotation.py

```diff
@@ -0,0 +1,498 @@
+
+    '''
+
+    return np.argmin(np.abs(vector_azimuth_time-azmuth_time_burst))
+
+
+@dataclass
+class BurstNoise: #For thermal noise correction
+    '''Noise correction information for Sentinel-1 burst'''
+    range_azimith_time: datetime.datetime = None
+    range_line: float = None
+    range_pixel: np.ndarray = None
+    range_lut: np.ndarray = None
+
+    azimuth_first_azimuth_line: int = None
+    azimuth_first_range_sample: int = None
+    azimuth_last_azimuth_line: int = None
+    azimuth_last_range_sample: int = None
+    azimuth_line: np.ndarray = None
+    azimuth_lut: np.ndarray = None
+
+    line_from: int = None
+    line_to: int = None
+
+
+    def from_noise_annotation(self, noise_annotation: NoiseAnnotation, azimuth_time: datetime.datetime,
+                              line_from: int, line_to: int, ipf_version: version.Version = version.parse('3.10')):
+        '''Extracts the noise correction information for individual burst from NoiseAnnotation
+
+        Parameters
+        ----------
+        noise_annotation: NoiseAnnotation
+            Subswath-wide noise annotation information
+        azimuth_time : datetime.datetime
+            Azimuth time of the burst
+        line_from: int
+            First line of the burst in the subswath
+        line_to: int
+            Last line of the burst in the subswath
+        ipf_version: float
+            IPF version of the SAFE data
+
+        '''
+
+        #threshold_ipf_version = 2.90 #IPF version that stared to provide azimuth noise vector
+        id_closest = closest_block_to_azimuth_time(noise_annotation.rg_list_azimuth_time, azimuth_time)
+        self.range_azimith_time = noise_annotation.rg_list_azimuth_time[id_closest]
+        self.range_line = noise_annotation.rg_list_line[id_closest]
+        self.range_pixel = noise_annotation.rg_list_pixel[id_closest]
+        self.range_lut = noise_annotation.rg_list_noise_range_lut[id_closest]
+
+        self.azimuth_first_azimuth_line = noise_annotation.az_first_azimuth_line
+        self.azimuth_first_range_sample = noise_annotation.az_first_range_sample
+        self.azimuth_last_azimuth_line = noise_annotation.az_last_azimuth_line
+        self.azimuth_last_range_sample = noise_annotation.az_last_range_sample
+
+        self.line_from = line_from
+        self.line_to = line_to
+
+        if ipf_version >= version_threshold_azimuth_noise_vector: 
+            #Azinuth noise LUT exists - crop to the extent of the burst
+            id_top = np.argmin(np.abs(noise_annotation.az_line-line_from))
+            id_bottom = np.argmin(np.abs(noise_annotation.az_line-line_to))
+            #put some margin when possible
+            if id_top > 0:
+                id_top -= 1
+            if id_bottom < len(noise_annotation.az_line)-1:
+                id_bottom += 1
+            self.azimuth_line = noise_annotation.az_line[id_top:id_bottom]
+            self.azimuth_lut = noise_annotation.az_noise_azimuth_lut[id_top:id_bottom]
```

Comment:
This excludes the pixel at `id_bottom`. Maybe you should add 1 to `id_bottom`? (please, review it carefully)

```suggestion
            self.azimuth_line = noise_annotation.az_line[id_top:id_bottom + 1]
            self.azimuth_lut = noise_annotation.az_noise_azimuth_lut[id_top:id_bottom + 1]
```

Reply:
Thanks @gshiroma for catching this. Committing your suggestion.